packages/graph/hash_graph/lib/graph/src/api/rest/data_type.rs

```diff
@@ -139,13 +140,19 @@ async fn get_latest_data_types<P: GraphPool>(
         ("uri" = String, Path, description = "The URI of the data type"),
     )
 )]
-async fn get_data_type<P: GraphPool>(
-    Path(uri): Path<VersionedUri>,
-    Extension(pool): Extension<Arc<P>>,
+async fn get_data_type<P: StorePool + Send>(
+    uri: Path<VersionedUri>,
+    pool: Extension<Arc<P>>,
 ) -> Result<Json<DataType>, StatusCode> {
-    read_from_store::<DataType, _, _, _>(pool.as_ref(), &uri)
-        .await
-        .map(Json)
+    read_from_store(
+        pool.as_ref(),
+        &DataTypeQuery::new()
+            .with_uri(uri.base_uri())
+            .with_version(uri.version()),
+    )
+    .await
+    .and_then(|mut data_types| data_types.pop().ok_or(StatusCode::NOT_FOUND))
```

Comment:
As mentioned in *Known issues*, `Read<T>::read` will now always return `Vec<T>`

Reply:
It should definitely be possible but doing this now is out of the scope of this PR:
> This could be resolved by adding a read_one method on the trait, which reuses the read method by default, but this is not required for now, so this was left out.

I will revisit this at a later point. I added a todo for this: ebdb86b44f4ab4bf298017c84cfa9a7ae90a1b25